### PR TITLE
[CARBONDATA-1686]  Presto Version Upgrade to  0.186

### PR DESCRIPTION
**Pull request by bhavya411**
- [No] Any interfaces changed?
 
 - [Yes ] Any backward compatibility impacted?  Carbondata will now work with Presto Server 0.186 onward
 
 - [ No] Document update required?

 - [ Yes] Testing done
       Since it is a version upgrade  all the integration test cases are executed successfuly.
       Performance testing done manually and there is improvement in query execution times.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/917/

File: integration/presto/src/main/java/org/apache/carbondata/presto/CarbondataPageSource.java [hunk 1]
```diff
@@ -78,10 +78,6 @@ private CarbondataPageSource(List<Type> types, RecordCursor cursor) {
     this.readers = createStreamReaders();
   }
 
-  @Override public long getTotalBytes() {
```
Comment (chenliang613) [inline integration/presto/src/main/java/org/apache/carbondata/presto/CarbondataPageSource.java]: upgrade presto version , why need to remove this method ?

File: integration/presto/src/main/java/org/apache/carbondata/presto/PrestoFilterUtil.java [hunk 2]
```diff
@@ -261,8 +261,6 @@ else if (type instanceof DecimalType) {
         return new BigDecimal(new BigInteger(String.valueOf(rawdata)),
             ((DecimalType) type).getScale());
       }
-    } else if (type.equals(TimestampType.TIMESTAMP)) {
```
Comment (chenliang613) [inline integration/presto/src/main/java/org/apache/carbondata/presto/PrestoFilterUtil.java]: this is PR1459 fixed TIMESTAMP issues.

**Comment by chenliang613** [comment]
@bhavya411 please rework this PR, there are many issues.

File: integration/presto/src/main/java/org/apache/carbondata/presto/CarbondataPageSource.java (hunk 1 above)
Comment (bhavya411) [reply integration/presto/src/main/java/org/apache/carbondata/presto/CarbondataPageSource.java]: The Presto 0.186 has remove this method from specification, so we have to remove it from our code as well

File: integration/presto/src/main/java/org/apache/carbondata/presto/PrestoFilterUtil.java (hunk 2 above)
Comment (bhavya411) [reply integration/presto/src/main/java/org/apache/carbondata/presto/PrestoFilterUtil.java]: This has been fixed

**Comment by bhavya411** [comment]
All issues have been resolved

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/928/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1532/

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/936/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1543/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1551/

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/943/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1558/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/956/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1571/

**Comment by bhavya411** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/964/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1580/

**Pull request closed without merging**